src/helpers.ts

```diff
@@ -214,10 +214,38 @@ async function getPyrightInfo(): Promise<NpmRegistryResponse> {
     return parseNpmRegistryResponse(JSON.parse(body));
 }
 
-function getPyrightVersion() {
+async function getPyrightVersion(): Promise<string> {
     const versionSpec = core.getInput("version");
     if (versionSpec) {
         return new SemVer(versionSpec).format();
     }
+
+    const pylanceVersion = core.getInput("pylance-version");
+    if (pylanceVersion) {
+        if (pylanceVersion !== "latest-release" && pylanceVersion !== "latest-prerelease") {
+            new SemVer(pylanceVersion); // validate version string
+        }
+
+        return await getPylancePyrightVersion(pylanceVersion);
+    }
+
     return "latest";
 }
+
+async function getPylancePyrightVersion(pylanceVersion: string): Promise<string> {
+    const client = new httpClient.HttpClient();
+    const resp = await client.get(
+        `https://raw.githubusercontent.com/microsoft/pylance-release/main/builds/${pylanceVersion}.json`,
```

Comment:
I don't mind if this is how you want to do it, but would it make sense to have a single file with the mappings again? The directory will end up growing indefinitely which sounds like it'd get unweildy in time.

Reply:
I hear what you're saying, but I think the pain of a large directory will be minimal, if not zero. Having one file per version simplifies the code that generates and consumes these files. Since the consumers (initially pyright-action and pyright-python) are all expected to be external, I think that simplification outweighs the possible pain.

I don't expect people to be looking at these files much, but if they do, the one downside I see is that GitHub only displays the first 1000 files in a directory. In the past 12 months we shipped ~110 releases. We're considering switching to nightly prerelease and monthly stable releases, and then I'm guessing we'll ship around 250 releases per year. If the 1000 file display limit becomes an issue, I would be fine with deleting, say, all of the files that are more than 2-3 years old. If someone is pinned on a version that old, I'm ok with breaking them. They should upgrade.

Reply:
No worries; I just asked because once this is in, it's there for good.